docs/sphinx/about/whats-new.txt

```diff
@@ -1,6 +1,112 @@
 Version history
 ===============
 
+5.2.0-m4 (2016 June 24)
+-----------------------
+
+Top-level Bio-Formats API changes:
+
+* Java 1.7 is now the minimum supported version
+* the native-lib-loader dependency has been bumped to version 2.1.4
+* all the ome.jxr classes have been deprecated to make clear that there is no
+  jeg-xr support implemented in Bio-Formats as yet
+* the DataTools API has been extended to add a number of utility functions to:
+   - account for decimal separators in different locales
+   - parse a String into Double, Flat, Integer etc
+   - handle NumberFormatException thrown when parsing Unit tests
+* the Logging API has been updated to respect logging frameworks
+  (log4j/logback) initialized via a binding-specific configurations file and
+  to prevent `DebugTools.enableLogging(String)` from overriding initialized
+  logger levels
+
+The Data Model version 2016-06 has been released to introduce
+`Folders <http://blog.openmicroscopy.org/data-model/future-plans/2016/05/23/folders-upcoming/>`_,
+and to simplify both the graphical aspects of the model and code generation.
+Full details will be available in the
+:model_doc:`OME Model and Formats Documentation <>` shortly. OME-XML changes
+include:
+
+* `Map` is now a complexType rather than an element and `MapPairs` has been
+  dropped
+* extended enum metadata has been introduced to better support units
+* `Shape` and `LightSource` are now complexTypes rather than elements
+* BinData has been added to codegen to handle raw binary data
+* various code generation improvements to allow the model to be implemented in
+  C++
```

Comment:
Ive tried to write something a few times and this is about as good as Ive come up with. Might not be best to go into too much detail.

Various code generation improvements to:
- Simplify and standardise the generation process
- Remove a number of hard coded exceptional cases allowing for easier maintenance and growth
- Allow for genuine abstract model types and enable C++ model implementation